runners/prism/java/src/main/java/org/apache/beam/runners/prism/ArtifactResolver.java

```diff
@@ -0,0 +1,110 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.prism;
+
+import static org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.base.Preconditions.checkState;
+
+import com.google.auto.value.AutoValue;
+import java.util.Optional;
+import org.apache.beam.model.pipeline.v1.RunnerApi;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.util.construction.DefaultArtifactResolver;
+import org.apache.beam.sdk.util.construction.PipelineTranslation;
+import org.apache.beam.sdk.util.construction.SdkComponents;
+
+/**
+ * The {@link ArtifactResolver} converts a {@link Pipeline} to a {@link RunnerApi.Pipeline} via
+ * resolving {@link RunnerApi.ArtifactInformation}.
+ */
+@AutoValue
+abstract class ArtifactResolver {
+
+  /**
+   * Instantiates a {@link ArtifactResolver} from the {@param pipeline}, applying defaults to the
+   * remaining dependencies.
+   */
+  static ArtifactResolver of(Pipeline pipeline) {
+    return ArtifactResolver.builder().setPipeline(pipeline).build();
+  }
+
+  static Builder builder() {
+    return new AutoValue_ArtifactResolver.Builder();
+  }
+
+  /**
+   * Converts the {@link #getPipeline()} using {@link PipelineTranslation#toProto} and {@link
+   * #getDelegate()}'s {@link
+   * org.apache.beam.sdk.util.construction.ArtifactResolver#resolveArtifacts}.
+   */
+  RunnerApi.Pipeline resolvePipelineProto() {
+    RunnerApi.Pipeline result = PipelineTranslation.toProto(getPipeline(), getSdkComponents());
+    return getDelegate().resolveArtifacts(result);
+  }
+
+  /**
+   * {@link ArtifactResolver} delegates to {@link
+   * org.apache.beam.sdk.util.construction.ArtifactResolver} to transform {@link
+   * RunnerApi.ArtifactInformation}. Defaults to {@link DefaultArtifactResolver#INSTANCE} if not
+   * set.
+   */
+  abstract org.apache.beam.sdk.util.construction.ArtifactResolver getDelegate();
+
+  /** The {@link Pipeline} from which {@link ArtifactResolver#resolvePipelineProto()}. */
+  abstract Pipeline getPipeline();
+
+  /**
+   * SDK objects that will be represented by {@link
+   * org.apache.beam.model.pipeline.v1.RunnerApi.Components}. Instantiated via {@link
+   * SdkComponents#create(PipelineOptions)} by default, where {@link PipelineOptions} are acquired
+   * from {@link #getPipeline}'s {@link Pipeline#getOptions}.
+   */
+  abstract SdkComponents getSdkComponents();
+
+  @AutoValue.Builder
+  abstract static class Builder {
+
+    abstract Builder setDelegate(
+        org.apache.beam.sdk.util.construction.ArtifactResolver artifactResolver);
```

Comment:
since there is a naming conflict, consider PrismArtifactResolver? As there is already ArtifactResolver(interface); DefaultArtifactResolver(an impl), and this follows the same naming convention